pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -31,47 +37,48 @@ type splitInstantQueryByIntervalMiddleware struct {
 
-	instantQuerySplittingMetrics
+	metrics instantQuerySplittingMetrics
 }
 
 type instantQuerySplittingMetrics struct {
 	splittingAttempts    prometheus.Counter
-	mappedSplitQueries   *prometheus.CounterVec
+	splittingSuccesses   prometheus.Counter
+	splittingSkipped     *prometheus.CounterVec
 	splitQueries         prometheus.Counter
 	splitQueriesPerQuery prometheus.Histogram
 }
 
-// Parsing evaluation result used in instantQuerySplittingMetrics
-const (
-	successKey = "success"
-	failureKey = "failure"
-	noopKey    = "noop"
-)
-
 func newInstantQuerySplittingMetrics(registerer prometheus.Registerer) instantQuerySplittingMetrics {
-	return instantQuerySplittingMetrics{
+	m := instantQuerySplittingMetrics{
 		splittingAttempts: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
-			Namespace: "cortex",
-			Name:      "frontend_instant_query_splitting_rewrites_attempted_total",
-			Help:      "Total number of instant queries the query-frontend attempted to split.",
+			Name: "cortex_frontend_instant_query_splitting_rewrites_attempted_total",
+			Help: "Total number of instant queries the query-frontend attempted to split by interval.",
+		}),
+		splittingSuccesses: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_frontend_instant_query_splitting_rewrites_succeeded_total",
+			Help: "Total number of instant queries the query-frontend successfully split by interval.",
 		}),
-		mappedSplitQueries: promauto.With(registerer).NewCounterVec(prometheus.CounterOpts{
-			Namespace: "cortex",
-			Name:      "frontend_instant_query_splitting_rewrites_succeeded_total",
-			Help:      "Number of instant queries the query-frontend attempted to split by evaluation type.",
-		}, []string{"evaluation"}),
+		splittingSkipped: promauto.With(registerer).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_frontend_instant_query_splitting_rewrites_skipped_total",
+			Help: "Total number of instant queries the query-frontend skipped or failed to split by interval.",
+		}, []string{"reason"}),
 		splitQueries: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
-			Namespace: "cortex",
-			Name:      "frontend_instant_query_split_queries_total",
-			Help:      "Total number of split partial queries.",
+			Name: "cortex_frontend_instant_query_split_queries_total",
+			Help: "Total number of split partial queries.",
 		}),
 		splitQueriesPerQuery: promauto.With(registerer).NewHistogram(prometheus.HistogramOpts{
-			Namespace: "cortex",
-			Name:      "frontend_instant_query_split_queries_per_query",
-			Help:      "Number of split partial queries a single instant query has been rewritten to.",
-			Buckets:   prometheus.ExponentialBuckets(2, 2, 10),
+			Name:    "cortex_frontend_instant_query_split_queries_per_query",
+			Help:    "Number of split partial queries a single instant query has been rewritten to.",
+			Buckets: prometheus.ExponentialBuckets(2, 2, 10),
 		}),
 	}
+
+	// Initialize known label values.
```

Comment:
Note to reviewers: this is a Prometheus best practice, so that metrics are exposed even if they have 0 value.